src/main/java/reposense/frontend/RepoSense.java

```diff
@@ -1,13 +1,39 @@
 package reposense.frontend;
 
+import java.io.File;
+import java.util.Date;
+import java.util.List;
+
+import reposense.dataobject.RepoConfiguration;
+import reposense.exception.ParseException;
+import reposense.parser.CliArgumentsParser;
+import reposense.parser.InputParameter;
+import reposense.report.RepoInfoFileGenerator;
+import reposense.system.CsvConfigurationBuilder;
 
 public class RepoSense {
 
     public static final int FLAG_SUCCESS = 0;
     public static final int FLAG_ERROR = 1;
 
     public static void main(String[] args) {
-        if (args.length == 0 || new Client().run(args) == FLAG_ERROR) {
+        if (args.length == 0) {
+            showHelpMessage();
+        }
+
+        try {
+            CliArgumentsParser cliArgumentsParser = new CliArgumentsParser();
+            InputParameter parameter = cliArgumentsParser.parse(args);
```

Comment:
hmm, is it a good idea to incorporate data members into parser though? Since parser suppose to be an input > output thing.

Reply:
Indeed in some cases it is meaningful to separate the parser from the parsed output. e.g., if the parsing is complicated, if it is used multiple times, if the produced output has a meaning such as a the AbstractSyntaxTree we get by parsing a program.
In this case the parse method is used one time only (like a constructor). We could even have done `new CliArgumentsParser(args)`  except we usually try to avoid putting logic in a constructor.
May be the trick is not to think of it as a parser. Name the class `CliArguments`? Does this read better?

```java
CliArguments cliArguments = new CliArguments();
cliArguments.set(args);

File configFile = cliArguments.getConfigFile();
...